source/vibe/db/postgresql/package.d

```diff
@@ -107,16 +107,32 @@ class __Conn : dpq2.Connection
     {
         import vibe.core.core;
 
+        // Connection socket should be non-blocking while waiting
         auto sock = this.socket();
```

Comment:
this is uneffective, vibe-core does not need to use this socket, so it would be better to remove it, to be fair the only reason for creating this new socket is to destroy it, maybe we could eliminate it complety. But I am not sure if sock.blocking == false is true for everyone, in all systems where I test it, it is already in non-blocking state.

Reply:
This Socket is need only for switching socket into non-blocking mode by Phobos's `Socket.blocking`.

Reply:
But my test shows, it is already non-blocking, just try it and you will see

Reply:
But some years ago I am tried it before write this package. And Postgres documentation avoids to name this as "connection with non-blocking socket". 

We should ask about that in Postgresql developers community, IMHO.

Reply:
It seems it is non blocking for ages, but if we want to be sure we could test it, but this will have some performance impact